### PR TITLE
Validate employer name for employed members

### PR DESCRIPTION
**Pull request by jessieay**
**WHY**: This is a required field in MI Bridges, so we need to collect
the info in order to drive applications successfully.

screen shot 2017-09-26 at 3 23 05 pm

### CODE REVIEW

**Comment by jessieay** [comment]
[Employment / Self-Employment in MI Bridges](https://trello.com/c/zJwUGNZg/336-employment-self-employment-in-mi-bridges)

File: app/steps/many_members_step.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+class ManyMembersStep < Step
+  step_attributes :members
+
+  def valid?
+    members.each do |member|
+      member.errors.clear
+      validate_household_member(member)
+    end
+
+    members.map(&:errors).all?(&:blank?)
```
Comment (jayroh) [inline app/steps/many_members_step.rb]: If the errors are not empty/blank I think we should bubble up the contents of those objects up to the step.

File: spec/steps/income_details_per_member_spec.rb [hunk 2]
```diff
@@ -1,17 +1,28 @@
...
       snap_application = create(:snap_application, members: members)
 
       step = IncomeDetailsPerMember.new(members: snap_application.members)
+      step.valid?
 
-      expect(step.working_members).to match_array([working, self_employed])
+      expect(step.members.last.errors[:employed_employer_name]).
+        to eq([])
+      expect(step.members.first.errors[:employed_employer_name]).
+        to eq(["Make sure to answer this question"])
```
Comment (jayroh) [inline spec/steps/income_details_per_member_spec.rb]: I think(?) the form objects automatically handle displaying errors, yeah? If we bubble up the errors to the step object itself I believe it'll handle it without us reaching down into the associated `members` objects to check for errors. From the step's perspective that would mean all errors would be accessible via the step object itself.

```ruby
expect(step.errors[:employed_employer_name]).
  to eq(["Make sure to answer this question"])
```

I'll pull this down and play with it a little bit to see if I'm wrapping my head around this completely. I might be missing a bit by just looking at the diff :).

Comment (jessieay) [reply spec/steps/income_details_per_member_spec.rb]: in order for the errors to show correctly on the form, it needs to be on a `member`

Comment (jayroh) [reply spec/steps/income_details_per_member_spec.rb]: Ah-ha! Ok, cool

Comment (jessieay) [reply spec/steps/income_details_per_member_spec.rb]: the reason it is bubbling up to the parent `step` is that we do this check inside the `validate` method

**Pull request merged**